### DISCUSSION TITLE
Native Python 3 SDK Release Date?

### DISCUSSION THREAD

**Discussion by SamsonHua** [iRobotEducation/create3_docs#72] [Questions] [answered]
Hello!

Just got my Create 3 a few days ago and super excited to play around with it!

My main intended usage is to get more comfortable with ROS2 with a physical robot, but I was curious when the native Python SDK would be released? One of the documents mentioned June but was curious if that was still the case or if there was a more detailed release date?

Will the Python native SDK allow access to the USB-C port on the Create 3?

Thank you! :)

**Comment by shamlian** [comment]
Hi!

I don't have more details for you on timing the offline Python SDK except that we are currently still testing and improving it. I'll put you on the list for a future beta test, if you're interested.

The offline Python SDK works using the iRobot Coding protocol over BLE, just like the web version, [python.irobot.com](http://python.irobot.com/), with the same syntax.

If you are looking to use Python with the Create 3 robot over USB, I would recommend [rclpy](https://github.com/ros2/rclpy).

Hope this helps!

**Reply by justinIRBT** [reply]
We have an example driving the Create with rclpy here https://github.com/iRobotEducation/create3_examples/tree/galactic/create3_examples_py

**Reply by SamsonHua** [reply]
Ok thanks! Was able to successfully get my robot working with the rclpy examples!

**Reply by shamlian** [reply]
Awesome!! Looking forward to hearing about your cool projects; we hope you enjoy using the robot!

**Reply by ghost** [reply]
I would love to me included in the beta for the native python_sdk!

**Reply by niebrandon** [reply]
@shamlian i would also love to be included in the python sdk beta program

**Reply by shamlian** [reply]
~~Those of you requesting invites to the Python SDK beta, if you are willing to sign an MNDA, please email education@irobot.com referencing this post, including your name, institution, and GitHub username, and we'll help as best we can.~~ Edit: This is no longer necessary

**Reply by mgperkins** [reply]
Hello, Please add me as well to the beta python SDK program. Thanks!

**Reply by shamlian** [reply]
@mgperkins please see above

**Comment by GOPINATH-GS4** [comment]
Hi, 
I signed the mNDA. However I am not associated with any organization, I am an individual planning to develop a telepresence robot, and I did not hear anything back on when I will get access, Is there a timeframe on when you will give access to those who signed?  If you are not providing access to individuals who are not affiliated with any organization, please let me know. I can switch to something else 
Thanks & Regards
GOPI

**Reply by shamlian** [reply]
I can see you've signed but something fell through the cracks on our end -- we're on it.

**Reply by shamlian** [reply]
Sorry for the delay; please check your email for an invitation.

**Comment by shamlian** [comment, marked as answer]
Thank you for your interest in a native, installable iRobot® Python SDK. Given the level of interest we've received, the team has worked hard to promptly deliver a public beta release to the community, [now available here](https://github.com/iRobotEducation/irobot-edu-python-sdk/). This is intended to complement the iRobot® Python Web Playground, already available at [python.irobot.com](http://python.irobot.com/).
